### ISSUE TITLE
SC1058 false positive in bash for valid "for" loop without "do"

### ISSUE THREAD

**Issue by E5ten** [koalaman/shellcheck#1694]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1058
- My shellcheck version (`shellcheck --version` or "online"): 0.7.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
for i in {1..10}; { printf '%s\n' "${i}"; }

```

#### Here's what shellcheck currently says:
```
for i in {1..10}; { printf '%s\n' "${i}"; }
^-- SC1073: Couldn't parse this for loop. Fix to allow more checks.
                  ^-- SC1058: Expected 'do'.
                  ^-- SC1072: Expected 'do'. Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:
No warning, in bash this syntax for for loops works.

**Comment by koalaman** [comment]
Kindly fixed by @gabrielelana, @sphaso and @Muzietto!
